browser/templates/browser/investigator.html

```diff
@@ -94,6 +114,47 @@
 
     </form>
 
+<script>
+
+var institution_search = $('#id_institution_search');
+var institution_container = $('#institution-results-container');
+
+$(document).ready(function() {
+    $(".dropdown-toggle").dropdown();
+    $('[data-toggle="tooltip"]').tooltip();
+});
+
+var selectInstitutionResult = function(id, name) {
+    institution_container.empty();
+    institution_search.val(name);
+}
+
+var searchPromise = null;
+
+institution_search.on('keyup', function(e) {
+    var query = $(this).val();
+
+    if (query.length == 0){
+        institution_container.empty();
+    } else {
+        if (searchPromise) {
+            clearTimeout(searchPromise);
+        }
+        searchPromise = setTimeout(function() {
+            $.ajax("{% url "institution-list" %}?name=" + query, {
```

Comment:
I think your also supposed to pass a csrf token in a ajax call but I am not positive so I would check that out.

Reply:
AFAIK, csrf token is needed only for POST requests. This seems like a GET request.

Reply:
Thanks @hashknot for pointing that out. @tjquinn1 , I am using GET request for fetching information about the name of the institutions. This call is not modifying any data in the application and does not contain any sensitive information. <br/> Therefore csrf token protection is not required for this call.

Reply:
I'm not sure about using Django template tags inside javascript. I remember Julia pointing out, tomorrow if this javascript is moved to a .js file and is linked in this html file, this particular line won't work.

Reply:
That's true, if this js would be moved to a static file then this won't work anymore. I would leave it like that though since changing it would make thing more complicated and it's not that likely that we'll take out the js anytime soon.